controllers/dataplaneipset_controller.go

```diff
@@ -0,0 +1,247 @@
+
+	instance := &dataplanev1beta1.DataPlaneIPSet{}
+	err := r.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers. Return and don't requeue.
+			return result, nil
+		}
+		return result, err
+	}
+
+	helper, err := helper.NewHelper(
+		instance,
+		r.Client,
+		r.Kclient,
+		r.Scheme,
+		r.Log,
+	)
+
+	if err != nil {
+		// helper might be nil, so can't use util.LogErrorForObject since it requires helper as first arg
+		r.Log.Error(err, fmt.Sprintf("unable to acquire helper for DataPlaneIPSet %s", instance.Name))
+		return result, err
+	}
+
+	defer func() {
+		if err != nil {
+			instance.Status.Conditions.MarkTrue(dataplanev1beta1.DataPlaneIPSetReadyCondition, dataplanev1beta1.DataPlaneIPSetReadyMessage)
+		}
+		// update the overall status condition if service is ready
+		err := helper.PatchInstance(ctx, instance)
+		if err != nil {
+			r.Log.Error(err, "Can't patch Instance")
+			_err = err
+			return
+		}
+	}()
+
+	if instance.DeletionTimestamp.IsZero() && controllerutil.AddFinalizer(instance, helper.GetFinalizer()) {
+		return ctrl.Result{}, err
+	}
+
+	if !instance.DeletionTimestamp.IsZero() {
+		return r.ReconcileDelete(ctx, instance, helper, req)
+	}
+
+	instance.InitCondition()
+	// Build AssignIPDetails
+	networks := instance.Spec.DataPlaneNetworks
+
+	if instance.Status.IPAddresses == nil {
+		instance.Status.IPAddresses = make(map[string]dataplanev1beta1.IPReservation)
+	}
+
+	//find the the openstacknet objects
+	for _, network := range networks {
+		if _, ok := instance.Status.IPAddresses[network.Name]; ok {
+			continue
+		}
+		osnet := &dataplanev1beta1.OpenStackNet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      network.Name,
+				Namespace: instance.Namespace,
+			},
+		}
+		err = r.Get(ctx, types.NamespacedName{Name: network.Name, Namespace: instance.Namespace}, osnet)
+		if err != nil {
+			r.Log.Error(err, "Network Not Found")
+			return ctrl.Result{}, err
+		}
+
+		aid := ipam.AssignIPDetails{}
+		aid.Reservelist = osnet.Status.Reservations
+		for _, subnet := range osnet.Status.Subnets {
+			if subnet.Name == network.SubnetName {
+				_, cidr, err := net.ParseCIDR(subnet.Cidr)
+				if err != nil {
+					return ctrl.Result{}, err
+				}
+				aid.IPNet = *cidr
+				aid.RangeStart = net.ParseIP(subnet.AllocationRange.AllocationStart)
+				aid.RangeEnd = net.ParseIP(subnet.AllocationRange.AllocationEnd)
+			}
+			ipReservation, updatedReservations, err := ipam.AssignIP(aid, net.ParseIP(network.FixedIP))
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			ipReservation.Gateway = subnet.Gateway
+			ipReservation.Routes = subnet.Routes
+
+			r.Log.Info("Got Reservations")
+			aid.Reservelist = updatedReservations
+			osnet.Status.Reservations = append(osnet.Status.Reservations, ipReservation)
```

Comment:
a different controller should not update other controllers CR status

Reply:
It works though and I did not find any references which suggests not to do it. Though I agree on the issues with custom backup and restore requirement we discussed yesterday which can be avoided.